cmd/vsphere-xcopy-volume-populator/e2e-tests/ansible/setup-vm.yml

```diff
@@ -0,0 +1,292 @@
+          - vsphere_username | mandatory
+          - vsphere_password | mandatory
+          - datacenter | mandatory
+          - datastore | mandatory
+          - network | mandatory
+        fail_msg: "A required vSphere environment variable is not set or empty"
+
+    - name: Resolve datastore name and MoID
+      ansible.builtin.shell: |
+        python3 -c "
+        import os, ssl, json, sys
+        from pyVim.connect import SmartConnect, Disconnect
+        from pyVmomi import vim
+
+        vsphere_verify_ssl = os.environ.get('VSPHERE_VERIFY_SSL', 'true').lower()
+        if vsphere_verify_ssl in ['false', 'no', '0']:
+            context = ssl._create_unverified_context()
+        else:
+            context = ssl.create_default_context()
+
+        vsphere_password = os.environ.get('VSPHERE_PASSWORD')
+        vsphere_host = '{{ vsphere_host }}'
+        vsphere_username = '{{ vsphere_username }}'
+        datastore_name = '{{ datastore }}'
+
+        try:
+            si = SmartConnect(host=vsphere_host, user=vsphere_username, pwd=vsphere_password, sslContext=context)
+            if not si:
+                print(json.dumps({'error': 'Failed to connect to vSphere'}), file=sys.stderr)
+                sys.exit(1)
+
+            content = si.content
+            datastore_obj = None
+            datastores = content.viewManager.CreateContainerView(
+                container=content.rootFolder,
+                type=[vim.Datastore],
+                recursive=True
+            ).view
+
+            for ds in datastores:
+                if ds.name == datastore_name or ds._moId == datastore_name:
+                    datastore_obj = ds
+                    break
+
+            if datastore_obj:
+                print(json.dumps({'name': datastore_obj.name, 'id': datastore_obj._moId}))
+            else:
+                print(json.dumps({'error': 'Datastore not found'}), file=sys.stderr)
+                sys.exit(1)
+
+            Disconnect(si)
+
+        except Exception as e:
+            print(json.dumps({'error': str(e)}), file=sys.stderr)
+            sys.exit(1)
+        "
+      environment:
+        VSPHERE_PASSWORD: "{{ vsphere_password }}"
+        VSPHERE_VERIFY_SSL: "{{ lookup('env', 'VSPHERE_VERIFY_SSL') | default('true') }}"
+      register: datastore_info_result
+      changed_when: false
+
+    - name: Set datastore facts
+      ansible.builtin.set_fact:
+        datastore_name: "{{ (datastore_info_result.stdout | from_json).name }}"
+        datastore_id: "{{ (datastore_info_result.stdout | from_json).id }}"
+      when: datastore_info_result.rc == 0
+
+    - name: Fail if datastore not found
+      ansible.builtin.fail:
+        msg: "Failed to resolve datastore: {{ datastore_info_result.stderr | from_json | default({'error': 'Unknown error'}) }}"
+      when: datastore_info_result.rc != 0
+
+
+    - name: Create VM
+      community.vmware.vmware_guest:
+        hostname: "{{ vsphere_host }}"
```

Comment:
according to the docs the username can use 'VMWARE_USER' env variable. I guess that is similar for all of the credentials parameters here. we can get rid of a lot of repeating fields here

Reply:
right. fixed